service/frontend/fx.go

```diff
@@ -222,6 +225,16 @@ func NamespaceCountLimitInterceptorProvider(
 	)
 }
 
+func NamespaceValidatorInterceptorProvider(
```

Comment:
Ideally, things should be provided from the same package where they are defined.

Reply:
Yes. But they all are there. I will move them out together later (after `Resource` removal).